src/test_workflow/perf_test/perf_single_node_cluster.py

```diff
@@ -0,0 +1,55 @@
+
+import os
+from contextlib import contextmanager
+
+from test_workflow.perf_test.perf_test_cluster import PerfTestCluster
+
+
+class PerfSingleNodeCluster(PerfTestCluster):
+    """
+    Represents a performance test cluster. This class deploys the opensearch bundle with CDK. Supports both single
+    and multi-node clusters
+    """
+    def __init__(self, bundle_manifest, config, stack_name, cluster_config, current_workspace):
+        assert cluster_config.is_single_node_cluster(), "Cluster is not a single node cluster"
+        work_dir = os.path.join(current_workspace, "opensearch-cluster", "cdk", "single-node")
+        super().__init__(bundle_manifest, config, stack_name, cluster_config, current_workspace, work_dir)
+
+    def create_endpoint(self, cdk_output):
+        scheme = "https://" if self.cluster_config.security else "http://"
+        private_ip = cdk_output[self.stack_name]["PrivateIp"]
+        public_ip = cdk_output[self.stack_name].get("PublicIp", None)
+        self.is_endpoint_public = public_ip is not None
+        host = public_ip if public_ip is not None else private_ip
+        if host is not None:
```

Comment:
Isn't this an error in the case where `host` is `None`? What should happen here? Default to localhost?

Reply:
Yep. Ideally this shouldn't happen. I'll remove the 'None' check.